src/test/java/org/jenkinsci/plugins/workflow/cps/steps/RestartingLoadStepTest.java

```diff
@@ -218,4 +226,134 @@ public void pauseInsideLoad() throws Exception {
+        story.then(r -> {
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p1");
+            p.setDefinition(new CpsFlowDefinition(
+                    "echo(/Pre-semaphore value is ${" + EXISTING_VAR_NAME + "}/)\n" +
+                    "semaphore('wait')\n" +
+                    "echo(/Post-semaphore value is ${" + EXISTING_VAR_NAME + "}/)", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            SemaphoreStep.waitForStart("wait/1", b);
+            r.assertLogContains("Pre-semaphore value is " + EXISTING_VAR_VALUE, b);
+        });
+        story.then(r -> {
+            WorkflowJob p = r.jenkins.getItemByFullName("p1", WorkflowJob.class);
+            WorkflowRun b = p.getBuildByNumber(1);
+            SemaphoreStep.success("wait/1", null);
+            r.waitForCompletion(b);
+            r.assertBuildStatus(Result.SUCCESS, b);
+            r.assertLogContains("Post-semaphore value is " + EXISTING_VAR_VALUE, b);
+        });
+    }
+
+    @Test
+    public void existingBindingsOnRestart() {
+        story.then(r -> {
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            r.jenkins.getWorkspaceFor(p).child("a.groovy").write(
+                    "def call(arg) {echo (/a ran on ${arg}/)}; this", null);
+            ScriptApproval.get().approveSignature("method groovy.lang.Binding getVariables");
+            r.jenkins.getWorkspaceFor(p).child("b.groovy").write(
+                    "def m(arg) {echo (/${this} binding=${binding.variables}/); a(/${arg} from b/)}; this", null);
+            p.setDefinition(new CpsFlowDefinition(
+                    "a = 0;" +
+                    "node {a = load 'a.groovy'};" +
+                    "echo (/Pre-semaphore value is ${" + EXISTING_VAR_NAME + "}/);" +
+                    "semaphore 'wait';" +
+                    "echo (/Post-semaphore value is ${" + EXISTING_VAR_NAME + "}/);" +
+                    "def b;" +
+                    "node {b = load 'b.groovy'};" +
+                    "echo (/${this} binding=${binding.variables}/);" +
+                    "b.m('value')", true));
+            WorkflowRun b = p.scheduleBuild2(0).getStartCondition().get();
+            SemaphoreStep.waitForStart("wait/1", b);
+            r.assertLogContains("Pre-semaphore value is " + EXISTING_VAR_VALUE, b);
+        });
+        story.then( r -> {
+            WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+            WorkflowRun b = p.getBuildByNumber(1);
+            SemaphoreStep.success("wait/1", null);
+            r.waitForCompletion(b);
+            r.assertBuildStatus(Result.SUCCESS, b);
+            r.assertLogContains("Post-semaphore value is " + EXISTING_VAR_VALUE, b);
+            r.assertLogContains("a ran on value from b", b);
+        });
+    }
+
+    @Test
+    public void conflictingBindingsOnRestart() throws Exception {
```

Comment:
yes, that's basically it. I was just writing out a test that explicitly shows what the behavior was with conflicting variables. But I think it doesn't really take much wrangling to figure out that behavior without this test.  I'll go ahead and remove it. Less code is better, since it's not testing any additional edge cases, like you said.